### ISSUE TITLE
Seq2Seq: why length of input sequences should be equal

### ISSUE THREAD

**Issue by mehrdadscomputer** [fchollet/keras#6807]
As I see more LSTM implementation on keras, it seems that the length of input sequences should be equal. for example in [addition_rnn](https://github.com/fchollet/keras/blob/master/examples/addition_rnn.py), we use padding to make length of sequences equal.
But as you now LSTM is able to learn sequences with different lengths. So why we use padding?

I asked this question generally (not about keras) which people says there should not be such equalization.
[https://stats.stackexchange.com/questions/282319/how-to-feed-time-series-with-different-length-into-lstm](https://stats.stackexchange.com/questions/282319/how-to-feed-time-series-with-different-length-into-lstm)
I appreciate if anyone explain this issue.

**Comment by mehrdadscomputer** [comment]
I asked this question on slak channel of keras. so I am closing the issue here.

**Comment by jinfagang** [comment]
Any result or thoughts about this? Or now APIs in Keras not need to using padding?

**Comment by mehrdadscomputer** [comment]
@jinfagang 
I didn't get exact answer but people advised me to use this features:
1. Masking layer
2. sample weight initialization alongside with padding

I did some exploration but results were not good when I added masking layer. finally I decided not to use masking and just normalized my data to have sequence lengths and so I lost part of training data.

**Comment by mehrdadscomputer** [comment]
@jinfagang 
I recommend to read this too:
http://machinelearningmastery.com/data-preparation-variable-length-input-sequences-sequence-prediction/

**Comment by jinfagang** [comment]
@mehrdadscomputer Thanks pal, nice help.

**Comment by student9292** [comment]
@mehrdadscomputer I have the same problem, but my situation is a bit different. I want to train with a little part of the whole data and than use the whole sequence/data in prediction. So I can not cut my data for prediction. Padding masking did not work good, what can I do ?
